test_async_load_databases is flaky

cidb - https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSBjaGVja19zdGFydF90aW1lID49IG5vdygpIC0gSU5URVJWQUwgMjQgSE9VUgogICAgQU5EIChoZWFkX3JlZiA9ICdtYXN0ZXInIEFORCBzdGFydHNXaXRoKGhlYWRfcmVwbywgJ0NsaWNrSG91c2UvJykpCiAgICBBTkQgdGVzdF9zdGF0dXMgIT0gJ1NLSVBQRUQnCiAgICBBTkQgKHRlc3Rfc3RhdHVzIExJS0UgJ0YlJyBPUiB0ZXN0X3N0YXR1cyBMSUtFICdFJScpIAogICAgQU5EIGNoZWNrX3N0YXR1cyAhPSAnc3VjY2VzcycKICAgIEFORCBwb3NpdGlvbih0ZXN0X25hbWUsICd0ZXN0X2FzeW5jX2xvYWRfZGF0YWJhc2VzJykgPiAwCk9SREVSIEJZIGNoZWNrX3N0YXJ0X3RpbWU=

```
File: test_async_load_databases/test.py:336 - in test_materialized_views_cascaded_multiple
    assert query("select * from test_mv.z Format CSV") == "42,2\n"
E   AssertionError: assert '42,0\n' == '42,2\n'
E     
E     - 42,2
E     ?    ^
E     + 42,0
E     ?    ^
```

```
E   helpers.client.QueryRuntimeException: Client failed! Return code: 82, stderr: Received exception from server (version 25.10.1):
E   Code: 82. DB::Exception: Received from 172.16.2.6:9000. DB::Exception: Database test_mv already exists.. Stack trace:
```

--- a/tests/integration/test_async_load_databases/test.py
+++ b/tests/integration/test_async_load_databases/test.py
@@ -142,6 +142,8 @@ def dependent_tables_assert():
 
 def test_dependent_tables(started_cluster):
     query = node1.query
+    query("drop database if exists a")
+    query("drop database if exists lazy")
     query("create database lazy engine=Lazy(10)")
     query("create database a")
     query("create table lazy.src (n int, m int) engine=Log")
@@ -257,6 +259,7 @@ def test_async_load_system_database(started_cluster):
 
 def test_materialized_views(started_cluster):
     query = node1.query
+    query("drop database if exists test_mv")
     query("create database test_mv")
     query("create table test_mv.t (Id UInt64) engine=MergeTree order by Id")
     query("create table test_mv.a (Id UInt64) engine=MergeTree order by Id")
@@ -279,6 +282,7 @@ def test_materialized_views(started_cluster):
 
 def test_materialized_views_cascaded(started_cluster):
     query = node1.query
+    query("drop database if exists test_mv")
     query("create database test_mv")
     query("create table test_mv.t (Id UInt64) engine=MergeTree order by Id")
     query("create table test_mv.a (Id UInt64) engine=MergeTree order by Id")
@@ -301,6 +305,7 @@ def test_materialized_views_cascaded(started_cluster):
 
 def test_materialized_views_cascaded_multiple(started_cluster):
     query = node1.query
+    query("drop database if exists test_mv")
     query("create database test_mv")
     query("create table test_mv.t (Id UInt64) engine=MergeTree order by Id")
     query("create table test_mv.a (Id UInt64) engine=MergeTree order by Id")
@@ -354,6 +359,7 @@ def test_materialized_views_replicated(started_cluster):
     for node in [node1, node2]:
         node.query(
             f"""
+            DROP DATABASE IF EXISTS test_mv;
             CREATE DATABASE test_mv;
             CREATE TABLE test_mv.test_table_H(id UInt32)
             ENGINE = ReplicatedMergeTree('/clickhouse/tables/test_table_H', '{nodenum}')

--- a/tests/integration/test_async_load_databases/test.py
+++ b/tests/integration/test_async_load_databases/test.py
@@ -335,7 +335,10 @@ def test_materialized_views_cascaded_multiple(started_cluster):
     query("insert into to_join values(42, 4200)")
 
     node1.restart_clickhouse()
-    query("insert into test_mv.t values(42)")
+    query("insert into test_mv.t values(42)", settings={
+        # Make sure that pushing to MVs are not reordered
+        "max_threads": 1
+    })
     assert query("select * from test_mv.a Format CSV") == "42\n"
     assert query("select * from test_mv.x Format CSV") == '"42"\n'
     assert query("select * from test_mv.z Format CSV") == "42,2\n"

